### DISCUSSION TITLE
Specified sleep period exceeds maximum

### DISCUSSION THREAD

**Discussion by Bondian** [Chimildic/goofy#243] [Помощь с ошибкой] [answered]
![Скриншот 23-03-2023 122359](https://user-images.githubusercontent.com/85184404/227139243-925dec8e-89e9-4c7a-a1a5-27b4164753d8.png)

Для понимания - скрин одного из выполнений.
Это длится с 21 марта и всегда указывается одно и тоже количество секунд паузы, равное суткам.
Когда-то раньше, когда активно экспериментировал с Гуфи, были превышения, но по истечении суток
 всё возобновляло работу. Теперь пауза всегда 86401 секунду при каждой попытке. Это бан?

**Comment by Chimildic** [comment, marked as answer]
Пауза появляется из-за спама запросами к Spotify за **короткий** промежуток времени. Продолжительность паузы дает Spotify. Обычно 1-10 секунд. Экстремально большие значения говорят, что Spotify что-то не нравится. 

Чтобы выйти из такого "временного бана" нужно прекратить любые запросы:
1. Удалите все триггеры
2. Подождите пока пройдет указанное время, пусть с запасом
3. Восстановите триггеры, желательно без пересечений по времени (такие варианты лучше [объединить в одну функцию](https://chimildic.github.io/goofy/#/best-practices?id=Экономика-триггеров))

> `runTasks_` со сбором истории восстановится сам, когда будет запущен любой код

Как возможная альтернатива, создать новый клиент на стороне [Spotify Dashboard](https://developer.spotify.com/dashboard/) и поменять config на новый.

**Comment by Bondian** [comment]
> Экстремально большие значения говорят, что Spotify что-то не нравится.

Странно. Ведь до этого всё работало как часы. Я не вмешивался на протяжении большого периода времени, пока на почту не стали приходить сообщения об этом спаме.
> Чтобы выйти из такого "временного бана" нужно прекратить любые запросы:

Сейчас займусь. По итогам отпишусь

**Comment by Bondian** [comment]
Помогло создание нового клиента на стороне Spotify Dashboard, так как даже через неделю бездействия бан не был снят. Соответственно обновил файл config.
На Spotify Dashboard пришлось заходить через ВПН. Из России не пустили.

**Comment by hahahahywel** [comment]
I am having the same problem. I hadn't made any changes at all but then all of my playlists stopped updating with the error listed above. I have made a new client on spotify dashboard and made a new client on google apps script but nothing works. The issue happens early in the morning and then tends to sort itself out during the day.

I have followed your instructions of waiting for the specified time to pass but this does not fix it as it will reoccur every morning. I have also reduced the frequency of some of the playlists to once an hour instead of every half hour which used to work fine before.

Any help is appreciated

Thanks
